service/history/hsm/tree.go

```diff
@@ -375,7 +432,8 @@ func (n *Node) AddChild(key Key, data any) (*Node, error) {
 	return node, nil
 }
 
-// DeleteChild deletes an immediate child node and all its descendants from the tree.
+// DeleteChild marks a child node and all its descendants as deleted, removing them from the cache. After deletion,
+// nodes will only see operations relevant to their subtree via prefix matching of paths.
```

Comment:
This comment is unclear to me. WDYM by nodes will only see operations?
I'd keep this comment geared towards users of the (soon to be deprecated) framework.

Reply:
// DeleteChild marks a child node and all its descendants as deleted, removing them from the cache. No transitions will
// be allowed after deleting a child.